### PR TITLE
[clang][modules] stdarg.h and stddef.h shouldn't directly declare anything

### PR DESCRIPTION
**Pull request by ian-twilightcoder**
stdarg.h and especially stddef.h are textual and so everything they declare gets precompiled into all of their clients' pcm files. They shouldn't directly declare anything though, their purpose is to select what submodules get imported, and not to add duplicate declarations to all of their clients. Make it so that they always ignore their header guards, even without modules, and declare them in separate header files so that they only go into the stdarg/stddef pcms. Still declare them in case clients rely on them.

### CODE REVIEW

File: clang-tools-extra/clang-include-fixer/find-all-symbols/STLPostfixHeaderMap.cpp [hunk 1]
```diff
@@ -15,9 +15,11 @@ const HeaderMapCollector::RegexHeaderMap *getSTLPostfixHeaderMap() {
   static const HeaderMapCollector::RegexHeaderMap STLPostfixHeaderMap = {
       {"include/__stdarg___gnuc_va_list.h$", "<cstdarg>"},
       {"include/__stdarg___va_copy.h$", "<cstdarg>"},
+      {"include/__stdarg_macro.h$", "<cstdarg"},
       {"include/__stdarg_va_arg.h$", "<cstdarg>"},
       {"include/__stdarg_va_copy.h$", "<cstdarg>"},
       {"include/__stdarg_va_list.h$", "<cstdarg>"},
+      {"include/__stddef_macro.h$", "<cstddef"},
```
Comment (Bigcheese) [inline clang-tools-extra/clang-include-fixer/find-all-symbols/STLPostfixHeaderMap.cpp:22]: Looks like the closing `>` is missing for these two.

**Review by Bigcheese — COMMENTED** [review]
I think I like this as a solution, although I wonder if __stdarg_guard_macro.h would be a better name. The name now somewhat implies that it contains the macros that stddef and stdarg define.

File: clang-tools-extra/clang-include-fixer/find-all-symbols/STLPostfixHeaderMap.cpp (hunk 1 above)
Comment (ian-twilightcoder) [reply clang-tools-extra/clang-include-fixer/find-all-symbols/STLPostfixHeaderMap.cpp:22]: Oh, dangit, thanks.

**Comment by ian-twilightcoder** [comment]
> I think I like this as a solution, although I wonder if __stdarg_guard_macro.h would be a better name. The name now somewhat implies that it contains the macros that stddef and stdarg define.

But the macros don't actually guard anything. _macro isn't the best name in the world though, maybe _header_macro.h is ok even though they aren't traditional "header macros"?

**Review by Bigcheese — APPROVED** [review]
lgtm

**Pull request merged**